MetaMask/metamask-extension
# Refactor Errors E2E test to no longer rely upon a real error

## Explanation
We were previously relying on a known bug to test our integration with Sentry (making sure sentry attempts to send errors to the server). This PR replaces that by triggering an error that will only exist in Test.


## Manual Testing Steps
1. Build test mode
2. In the console, type 'stateHooks.throwTestError()'
3. View the network tab and see that it was sent to sentry (or would have been)
4. Also check E2E test works.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
